source/vision/visionHandler.py

```diff
@@ -231,8 +251,14 @@ def initializeProvider(
 				raise exceptions.ProviderInitException(
 					f"Trying to initialize provider {providerId} which reported being unable to start"
 				)
-			# Initialize the provider.
-			providerInst = providerCls()
+			try:
+				# Initialize the provider.
+				providerInst = providerCls()
+			except Exception as e:
+				# Disable the provider, so that it does not error every startup.
+				providerCls.enableInConfig(False)

```

Comment:
Why is this? Note that for braille display drivers and synth drivers, they aren't removed from the config when they fail starting. I think that's ok. Imagine the case where the screen curtain fails starting once and is then removed from the config without user visible popup. I don't think that's what people really want.

Reply:
Vision providers can, and should use the "can start" mechanism to report temporary failures. An exception during initialization is an unexpected failure, quite possibly causing instability. My assumption is that in this case it's more likely to be a permanent failure for the provider, some case that the developer didn't consider. Further I figure instability is worse than having to re-enable the provider after the issue is fixed.